app/scripts/modules/core/src/managed/ManagedResourceObject.tsx

```diff
@@ -21,23 +22,50 @@ const kindIconMap: { [kind: string]: IconNames } = {
   'application-load-balancer': 'loadBalancer',
 };
 
-function getIconTypeFromKind(kind: string) {
-  return kindIconMap[getKindName(kind)] ?? 'placeholder';
-}
+const getIconTypeFromKind = (kind: string) => kindIconMap[getKindName(kind)] ?? 'placeholder';
+
+const getResourceName = ({ moniker: { app, stack, detail } }: IManagedResourceSummary) =>
+  [app, stack, detail].filter(Boolean).join('-');
```

Comment:
If this is common enough, then I can move this to `ManagedReader` as a utility too!

Reply:
yeah that's a good point. I have no opinion on whether we need to do it _right now_ (there are 2 existing usages, this will be the third), but if we did I bet we could figure out a nice way to tie it in with [our existing name utilities](https://github.com/spinnaker/deck/tree/master/app/scripts/modules/core/src/naming). I don't love the way the existing ones are structured/named (ironically), but it does seem like the right place for both the kind name extraction that lives in `ManagedReader` and this (there's an existing implementation called `getClusterName()` which is definitely aging poorly and needs a rename/refactor).